lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -397,24 +384,33 @@ def student_submissions(course, all_students=False):
             for unit in subsection.get_children():
                 unit_name = unit.display_name_with_default
 
-                row = [section_name.encode('utf-8'), subsection_name.encode('utf-8'), unit_name.encode('utf-8'), '']
+                parent_metadata = [section_name, subsection_name, unit_name]
                 for component in unit.get_children():
                     if component.category == 'problem':
-                        row[-1] = component.display_name_with_default.encode('utf-8')
-                        modules = StudentModule.objects.filter(course_id=course.id, grade__isnull=False, module_state_key=component.location)
-                        if modules:
-                            answers = [''] * len(students)
-                            for module in modules:
-                                try:
-                                    state_dict = json.loads(module.state) if module.state else {}
-                                    raw_answers = state_dict.get("student_answers", {})
-                                except ValueError:
-                                    log.error("Student submissions: Could not parse module state for " +
-                                              "StudentModule id={}, course={}".format(module.id, course.id))
-                                    continue
-                                pretty_answers = u', '.join(u"{problem}={answer}".format(problem=problem, answer=answer) for (problem, answer) in raw_answers.items())
-                                answers[student_cols[module.student.username]] = pretty_answers.encode('utf-8')
-                            if any(answers):
-                                datarows.append(row + answers)
-
-    return header, datarows
+                        yield component, parent_metadata
+
+
+def student_submissions(course):
+    """
+    Yields student submissions for all problems in course for writing out to a CSV file.
+    """
+    order = 1
+    for problem_component, parent_metadata in iterate_problem_components(course):
+        problem_component_info = parent_metadata + [problem_component.display_name_with_default, order, problem_component.location]
+        modules = StudentModule.objects.filter(course_id=course.id, grade__isnull=False, module_state_key=problem_component.location).order_by('student__username')
```

Comment:
Nit: This line is pretty crazy long (>160 chars). It'd be nice to see this slightly smaller:
```python
modules = StudentModule.objects.filter(
    course_id=course.id,
    grade__isnull=False,
    module_state_key=problem_component.location,
).order_by('student__username')
```